kafka-ui-react-app/src/components/Topics/Details/Messages/MessageItem.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { format } from 'date-fns';
+import JSONTree from 'react-json-tree';
+import { TopicMessage } from 'generated-sources';
+
+interface MessageItemProp {
+  partition: TopicMessage['partition'];
+  offset: TopicMessage['offset'];
+  timestamp: TopicMessage['timestamp'];
+  content: Record<string, unknown>;
+}
+
+const MessageItem: React.FC<MessageItemProp> = ({
+  partition,
+  offset,
+  timestamp,
+  content,
+}) => {
+  const getFormattedDate = (date: Date) => {
+    if (!date) return null;
+    return format(date, 'yyyy-MM-dd HH:mm:ss');
+  };
+
+  const getMessageContentBody = (messageContent: Record<string, unknown>) => {
```

Comment:
In case you do decomposition I would suggest describing a new component for content.

Reply:
Or get rid of this function and use 
```
<td style={{ width: 200 }}>
  {date ? format(date, 'yyyy-MM-dd HH:mm:ss') : null}
</td>
```

Reply:
> In case you do decomposition I would suggest describing a new component for content.
@aMoRRoMa do you mean move the body of `getMessageContentBody` function to the new component, e.g, MessageContent?

Reply:
yes